MonoDevelop.MSBuild.Editor/Completion/MSBuildCompletionCommitManager.cs

```diff
@@ -60,6 +89,29 @@ public CommitResult TryCommit (IAsyncCompletionSession session, ITextBuffer buff
 			return CommitResult.Unhandled;
 		}
 
+		private static void ShiftCaret (IAsyncCompletionSession session, int len, CaretDirection caretDirection)
+		{
+			switch (caretDirection) {
+			case CaretDirection.Left:
+				for (int i = 0; i < len; i++) {
+					session.TextView.Caret.MoveToPreviousCaretPosition ();
+				}
+				return;
+			case CaretDirection.Right:
+				for (int i = 0; i < len; i++) {
+					session.TextView.Caret.MoveToNextCaretPosition ();
+				}
+				return;
+			case CaretDirection.Top:
+				//TO DO
```

Comment:
Add link to issue describing what needs to be done

Reply:
TBH I can't think of a case where we'd want to move the caret up/down, so I think it would be fine to throw an ArgumentException.

Reply:
`throw new ArgumentException ($"Unsupported value '{caretDirection}'")` has been added

Reply:
If this isn't going to be implemented, consider removing the `TO DO` comments and merging the two identical `case` blocks.